### PR TITLE
Use identifiers.js as single source of truth

### PR DESCRIPTION
**Pull request by jonandernovella**
**Related issue(s) and PR(s)**  
This PR contributes to https://github.com/MetabolicAtlas/MetabolicAtlas/issues/1038

**Type of change**  

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other
 
**List of changes made**  

- Merged two files and used better names for the objects

### CODE REVIEW

**Comment by mihai-sysbio** [comment]
@jon could you please use the conventional commits conventions as mentioned in the [NBIS Development Guidelines ](https://github.com/NBISweden/development-guidelines#helpful-commit-messages) (screenshot below):

image

**Review by perjo — APPROVED** [review]

**Comment by perjo** [comment]
Looks good and tested together with https://github.com/MetabolicAtlas/MetabolicAtlas/pull/1306 without errors.

**Comment by jonandernovella** [comment]
> @jon could you please use the conventional commits conventions as mentioned in the [NBIS Development Guidelines ](https://github.com/NBISweden/development-guidelines#helpful-commit-messages) (screenshot below):
> 
> image

These are simply tips. As we discussed earlier, developers are encouraged but never obliged to follow any rules in my workplace.

**Review by mihai-sysbio — COMMENTED** [review]
While the `identifiers.js` content is the merger of two different files, it's not quite yet a single source of truth. For example, `KEGG.compound` is still defined twice (but now in one file rather than in two).

**Comment by mihai-sysbio** [comment]
Further clarification with @kusalananda's help: the aim is "no repetition".

File: parser.js [hunk 1]
```diff
@@ -247,7 +247,18 @@ const getComponentExternalDb = (
         ) {
           continue;
         }
-        const dbName = dbIdentifiersDict[fcomponent]["dbname_map"][header];
+        const crossReferencesArray = Object.values(crossReferencesDict);
+        const referenceData = crossReferencesArray.find((item) =>
+          item.headers.includes(header)
+        );
+        const dbName = referenceData ? referenceData.name : "";
```
Comment (mihai-sysbio) [inline parser.js:254]: I haven't tested this yet, so just by looking I would think:
```suggestion
        const dbName = referenceData ? referenceData.db : "";
```
or perhaps even nicer to rename `.db` to something nicer such as `.dbName`

**Review by pdmnyberg — APPROVED** [review]
Seems to behave as expected after the latest change. LGTM. Good work

**Pull request merged**